Stix Parser Enchasement

daryakoval opened the pull request:
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-577

## Description
Stix parser new version. 
## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [x] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

dantavori commented on Packs/CommonScripts/Scripts/CreateIndicatorsFromSTIX/CreateIndicatorsFromSTIX.py:
```diff
@@ -1,69 +1,71 @@
     errors = list()
     for indicator in indicators:
+        indicator['type'] = indicator.get('indicator_type')
+        relationship_object = create_relationships_from_entity(indicator.get('relationships'))
```
please check with Dror if we want to add a `create_relationships` boolean arg. We do it in feed integrations, sounds legit to add it here

daryakoval commented on Packs/Base/Scripts/StixParser/StixParser.py:
```diff
@@ -63,351 +131,1460 @@ def convert_to_json(string):
+        Polls the taxii server and builds a list of cortex indicators objects from the result
+        :return: Cortex indicators list
+        """
+        if js_content.get('id', '').startswith('bundle--') and not js_content.get('objects'):
```
changed it. not relevant

daryakoval commented on Packs/Base/Scripts/StixParser/StixParser.py:
```diff
@@ -63,351 +131,1460 @@ def convert_to_json(string):
+        for obj_type, stix_objects in envelopes.items():
+            if obj_type != "relationship":
+                for obj in stix_objects:
+                    # we currently don't support extension object
+                    if obj.get('type') == 'extension-definition':
+                        continue
```
the line: `parse_objects_func[obj_type](obj)` will fail since no function for it in the dictionary, but this type is legal for stix.

daryakoval commented on Packs/Base/Scripts/StixParser/StixParser.py:
```diff
@@ -63,351 +131,1460 @@ def convert_to_json(string):
+            b_stixid = relationships_object.get('target_ref', '')
+            b_object = self.parsed_object_id_to_object.get(b_stixid, {})
+
+            if not a_object or not b_object:
```
It should not happen, but better to check it and avoid unnecessary failures.

daryakoval commented on Packs/Base/Scripts/StixParser/StixParser.py:
```diff
@@ -63,351 +131,1460 @@ def convert_to_json(string):
+            if not a_object or not b_object:
+                continue
+
+            a_value, a_type = a_object.get('value'), a_object.get('indicator_type')
+            b_value, b_type = b_object.get('value'), b_object.get('indicator_type')
```
In TAXII2 we create `$dummy_indicator` in order to create relationships... This option will not work when we are not use `demisto.createIndicators()`

daryakoval commented:
@ShirleyDenkberg please review only the following files: 
Packs/Base/ReleaseNotes/1_31_28.md
Packs/CommonScripts/ReleaseNotes/1_9_2.md
Packs/CommonScripts/Scripts/CreateIndicatorsFromSTIX/README.md
Packs/Base/Scripts/StixParser/README.md
Packs/CommonScripts/Scripts/CreateIndicatorsFromSTIX/CreateIndicatorsFromSTIX.yml
Packs/Base/Scripts/StixParser/StixParser.yml

ShirleyDenkberg commented:
@dantavori Doc review completed.

dantavori reviewed (approved):
great job

daryakoval commented:
@ShahafBenYakir 
Force merge needed - I rewrite the script and did the following changes that fails the validations:
1. I deleted `TestData` folder and its files and added new tests instead. 
2.  unit tests coverage is currently 77.88, but was 85.7. I removed the previous short code and wrote a new one (longer in 1500 lines) so the coverage is down for this reason.
